components/director/internal/domain/labeldef/service.go

```diff
@@ -107,16 +108,18 @@ func (s *service) Update(ctx context.Context, def model.LabelDefinition) error {
 	return nil
 }
 
-// TODO: Add deleting related labels logic
 func (s *service) Delete(ctx context.Context, tenant, key string, deleteRelatedLabels bool) error {
-	if deleteRelatedLabels == true {
-		return errors.New("deleting related labels is not yet supported")
-	}
-
 	if key == model.ScenariosKey {
 		return fmt.Errorf("Label Definition with key %s can not be deleted", model.ScenariosKey)
 	}
 
+	if deleteRelatedLabels {
```

Comment:
This if should be moved to the bottom - we need to check if LabelDefinition exists, right?

Reply:
done